diff -up SOGo-5.12.1/SOPE/GDLContentStore/GCSOpenIdFolder.h.omv~ SOGo-5.12.1/SOPE/GDLContentStore/GCSOpenIdFolder.h
--- SOGo-5.12.1/SOPE/GDLContentStore/GCSOpenIdFolder.h.omv~	2025-05-27 20:59:21.392155064 +0200
+++ SOGo-5.12.1/SOPE/GDLContentStore/GCSOpenIdFolder.h	2025-05-27 20:59:26.532890824 +0200
@@ -19,7 +19,7 @@
  */
 
 #ifndef GCSOPENIDFOLDER_H
-#define GCSOPENIDOLDER_H
+#define GCSOPENIDFOLDER_H
 
 
 @class NSCalendarDate;
diff -up SOGo-5.12.1/SoObjects/SOGo/SOGoOpenIdSession.m.omv~ SOGo-5.12.1/SoObjects/SOGo/SOGoOpenIdSession.m
--- SOGo-5.12.1/SoObjects/SOGo/SOGoOpenIdSession.m.omv~	2025-05-27 21:00:38.827220326 +0200
+++ SOGo-5.12.1/SoObjects/SOGo/SOGoOpenIdSession.m	2025-05-27 21:01:03.874135918 +0200
@@ -82,7 +82,7 @@ static BOOL SOGoOpenIDDebugEnabled = YES
         sendDomainInfo = [domainInfo boolValue];
 
       
-    userTokenInterval        = [_config objectForKey: @"SOGoOpenIdTokenCheckInterval"];
+    userTokenInterval        = [_config integerForKey: @"SOGoOpenIdTokenCheckInterval"];
     [self _loadSessionFromCache: forDomain];
 
     if(cacheUpdateNeeded)
