diff -up SOGo-5.12.10/SoObjects/Mailer/NSData+SMIME.m.openssl SOGo-5.12.10/SoObjects/Mailer/NSData+SMIME.m
--- SOGo-5.12.10/SoObjects/Mailer/NSData+SMIME.m.openssl
+++ SOGo-5.12.10/SoObjects/Mailer/NSData+SMIME.m
@@ -38,6 +38,7 @@
 #include <openssl/x509.h>
 #include <openssl/x509v3.h>
 #include <openssl/pkcs12.h>
+#include <openssl/asn1.h>
 #include <openssl/pem.h>
 #endif
 
@@ -594,7 +595,8 @@
 STACK_OF(X509_ALGOR) *CMS_get_smimecap(CMS_SignerInfo *si)
 {
     X509_ATTRIBUTE *attr;
-    ASN1_TYPE *cap;
+    const ASN1_TYPE *cap;
+    const ASN1_STRING *seq;
     const unsigned char *p;
 
     attr = CMS_signed_get_attr(si, CMS_signed_get_attr_by_NID(si, NID_SMIMECapabilities, -1));
@@ -603,9 +605,10 @@
     cap = X509_ATTRIBUTE_get0_type(attr, 0);
     if (!cap || (cap->type != V_ASN1_SEQUENCE))
        return NULL;
-    p = cap->value.sequence->data;
+    seq = cap->value.sequence;
+    p = ASN1_STRING_get0_data(seq);
     return (STACK_OF(X509_ALGOR) *)
-        ASN1_item_d2i(NULL, &p, cap->value.sequence->length,
+        ASN1_item_d2i(NULL, &p, ASN1_STRING_length(seq),
                       ASN1_ITEM_rptr(X509_ALGORS));
 }
 
