diff -up SOGo-5.12.3/SOPE/NGCards/iCalYearlyRecurrenceCalculator.m.4~ SOGo-5.12.3/SOPE/NGCards/iCalYearlyRecurrenceCalculator.m
--- SOGo-5.12.3/SOPE/NGCards/iCalYearlyRecurrenceCalculator.m.4~	2025-07-12 09:48:08.000000000 +0200
+++ SOGo-5.12.3/SOPE/NGCards/iCalYearlyRecurrenceCalculator.m	2025-07-16 01:00:49.889850155 +0200
@@ -45,7 +45,7 @@
   NSMutableArray *ranges;
   NSArray *byMonth;
   NSCalendarDate *firStart, *firEnd, *lastStartDate, *lastEndDate, *rStart, *rEnd, *until, *referenceDate, *rTemp;
-  NSInteger *hoursOfOffset;
+  NSInteger hoursOfOffset;
   iCalMonthlyRecurrenceCalculator *monthlyCalc;
   unsigned j, yearIdxInRange, numberOfYearsInRange, count, interval, monthDiff;
   int diff, repeatCount, currentMonth, origNbDaysInMonth;
diff -up SOGo-5.12.3/SoObjects/Mailer/SOGoMailFolder.m.4~ SOGo-5.12.3/SoObjects/Mailer/SOGoMailFolder.m
--- SOGo-5.12.3/SoObjects/Mailer/SOGoMailFolder.m.4~	2025-07-12 09:48:08.000000000 +0200
+++ SOGo-5.12.3/SoObjects/Mailer/SOGoMailFolder.m	2025-07-16 01:00:10.895278180 +0200
@@ -104,10 +104,11 @@ static NSComparisonResult _compareFetchR
   return [modseq1 compare: modseq2];
 }
 
-static NSInteger _compareFetchResultsByUID (id entry1, id entry2, NSDictionary *uids)
+static NSComparisonResult _compareFetchResultsByUID (id entry1, id entry2, void *uidsp)
 {
   NSString *uid1, *uid2;
   NSUInteger pos1, pos2;
+  NSDictionary *uids=(NSDictionary*)uidsp;
 
   uid1 = [entry1 objectForKey: @"uid"];
   uid2 = [entry2 objectForKey: @"uid"];
diff -up SOGo-5.12.3/SoObjects/SOGo/SOGoMailer.h.4~ SOGo-5.12.3/SoObjects/SOGo/SOGoMailer.h
--- SOGo-5.12.3/SoObjects/SOGo/SOGoMailer.h.4~	2025-07-12 09:48:08.000000000 +0200
+++ SOGo-5.12.3/SoObjects/SOGo/SOGoMailer.h	2025-07-16 01:00:10.895400401 +0200
@@ -37,7 +37,7 @@
 {
   NSString *mailingMechanism;
   NSString *smtpServer;
-  BOOL *smtpMasterUserEnabled;
+  BOOL smtpMasterUserEnabled;
   NSString *smtpMasterUserUsername;
   NSString *smtpMasterUserPassword;
   NSString *authenticationType;
